### ISSUE TITLE
userLoaded fires before claims are ready

### ISSUE THREAD

**Issue by GordonBlahut** [perarnborg/vuex-oidc#94]
When oidc-client-js does a silent renew via iframe, it also calls the userinfo endpoint (depending on the configuration of course) which refreshs the claims. That's great because it ensures claims are frequently updated. 

But when the userLoaded event fires after a silent renew, that userinfo endpoint call may not have completed yet so while handling the event, the user's claims may be outdated.

Any idea how to wait for the claims to be updated as well?

**Comment by perarnborg** [comment]
I need to look into this before I can  answer.

**Comment by olkkoo** [comment]
I think I faced almost the same situation. I'm not sure about what silent renew or iframe have to do with this.

On my App.vue, I have

```
methods: {
    userLoaded: function (e) {
        console.log('I am listening to the user loaded event in vuex-oidc', e.detail)
        console.log('UserLoaded: ', store.state.oidcStore.user)
    }
}

mounted () {
    window.addEventListener('vuexoidc:userLoaded', this.userLoaded)
}
```

When I'm redirected back from identity provider, the event is raised at some point when the user obviously should be loaded. However, `store.state.oidcStore.user` remains null.

I would assume that user was already saved to store when `userLoaded` event was fired, but that is not the case.

**Comment by perarnborg** [comment]
@olkkoo I understand that this is confusing. The event is dispatched from oidc-client, and it will be triggered before the user is set in the store by vuex-oidc.So while the user is null in the store in your case it is just about to be set to the value of e.detail.

Perhaps I should look into changing this.

@GordonBlahut This was reported a while ago, but do you remember if this could be the same issue that you were having – that the new claims where not yet saved in the (but they might be present in the event payload (e.detail)?

**Comment by GordonBlahut** [comment]
It doesn't sound like the same issue I was having. 

My issue was due to claims being refreshed after `userLoaded` fires due to the nature of `oidc-client`. I think I worked around it by just not checking any claims in `userLoaded`. Instead I just made an observable wrapper around claims so that I could react to claims changes/update UI.

I haven't run into any situations where the user not being in the store during `userLoaded` was an issue since the user is available from the event anyway.
